tests/functional/test_plugins_config.py

```diff
@@ -10,299 +10,246 @@
-                'Provider {} config "{}" is "{}" while expected "{}"'.format(pname, k, str(actual_cnf[k]), str(v)),
+def two_providers(sdk_client_fs: ADCMClient) -> Tuple[Provider, Provider]:
+    """Get two providers with two hosts"""
+    uploaded_bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, "provider"))
+    first_provider, second_provider, *_ = [uploaded_bundle.provider_create(name=name) for name in ('first', 'second')]
+    providers = (first_provider, second_provider)
+    for provider in providers:
+        for suffix in ('first', 'second'):
+            provider.host_create(fqdn=f'{provider.name}-{suffix}')
+    return providers
+
+
+@pytest.mark.parametrize(
+    ('change_action_name', 'object_to_be_changed', 'action_owner'),
+    generate_cluster_success_params(action_prefix='change', id_template='change_{}_config'),
+)
+@pytest.mark.usefixtures("two_clusters")
+def test_cluster_related_objects(
+    change_action_name: str,
+    object_to_be_changed: Tuple[str, ...],
+    action_owner: Tuple[str, ...],
+    sdk_client_fs: ADCMClient,
+):
+    """
+    Check that correct multi_state_set calls can run successfully
+        with Cluster, Service, Component
+        and change only the objects that must be changed
+    Then unset state and check that none of objects has multi state
+    """
+    _test_successful_multi_state_set_unset(
+        change_action_name,
+        object_to_be_changed,
+        action_owner,
+        sdk_client_fs,
+        get_cluster_related_object,
+    )
+
+
+@pytest.mark.parametrize(
+    ('change_action_name', 'object_to_be_changed', 'action_owner'),
+    generate_provider_success_params(action_prefix='change', id_template='change_{}_config'),
+)
+@pytest.mark.usefixtures("two_providers")
+def test_provider_related_objects(
+    change_action_name: str,
+    object_to_be_changed: Tuple[str, ...],
+    action_owner: Tuple[str, ...],
+    sdk_client_fs: ADCMClient,
+):
+    """
+    Check that correct multi_state_set calls can run successfully
+        with Provider and Host
+        and change only the objects that must be changed
+    Then unset state and check that none of objects has multi state
+    """
+    _test_successful_multi_state_set_unset(
+        change_action_name,
+        object_to_be_changed,
+        action_owner,
+        sdk_client_fs,
+        get_provider_related_object,
+    )
+
+
+def test_host_from_provider(two_providers: Tuple[Provider, Provider], sdk_client_fs: ADCMClient):
+    """Change host config from provider"""
+    provider = two_providers[0]
+    host = provider.host()
+    provider_name = compose_name(provider)
+    host_name = compose_name(host)
+    with check_config_changed(sdk_client_fs, {host}), allure.step(
+        f'Set config of {host_name} with action from {provider_name}'
+    ):
+        run_provider_action_and_assert_result(provider, 'change_host_from_provider', config={'host_id': host.id})
+
+
+# pylint: disable-next=possibly-unused-variable
```

Comment:
Just curious what causes `possibly-unused-variable`?

Reply:
In the previous version, iteration was over ('cluster', 'service' etc.), not over (cluster, service)
So `service` was possibly unused variable
Guess now I can remove it